cloud/azure/azure_rm_networkinterface.py

```diff
@@ -0,0 +1,619 @@
+              default security group will be created.
+        aliases:
+            - security_group
+        required: false
+        default: null
+    open_ports:
+        description:
+            - When a default security group is created for a Linux host a rule will be added allowing inbound TCP
+              connections to the default SSH port 22, and for a Windows host rules will be added allowing inbound
+              access to RDP ports 3389 and 5986. Override the default ports by providing a list of open ports.
+        type: list
+        required: false
+        default: null
+    tags:
+        description:
+            - "Dictionary of string:string pairs to assign as metadata to the object. Metadata tags on the object
+              will be updated with any provided values. To remove tags use the purge_tags option."
+        required: false
+    purge_tags:
+        description:
+            - Use to remove tags from an object. Any tags not found in the tags parameter will be removed from
+              the object's metadata.
+        default: false
+        required: false
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Chris Houseknecht (@chouseknecht)"
+    - "Matt Davis (@nitzmahone)"
+'''
+
+EXAMPLES = '''
+    - name: Create a network interface with minimal parameters
+        azure_rm_networkinterface:
+            name: nic001
+            resource_group: Testing
+            virtual_network_name: vnet001
+            subnet_name: subnet001
+
+     - name: Create a network interface with private IP address only (no Public IP)
+        azure_rm_networkinterface:
+            name: nic001
+            resource_group: Testing
+            virtual_network_name: vnet001
+            subnet_name: subnet001
+            public_ip: no
+
+    - name: Create a network interface for use in a Windows host (opens RDP port) with custom RDP port
+        azure_rm_networkinterface:
+            name: nic002
+            resource_group: Testing
+            virtual_network_name: vnet001
+            subnet_name: subnet001
+            os_type: Windows
+            rdp_port: 3399
+
+    - name: Create a network interface using existing security group and public IP
+        azure_rm_networkinterface:
+            name: nic003
+            resource_group: Testing
+            virtual_network_name: vnet001
+            subnet_name: subnet001
+            security_group_name: secgroup001
+            public_ip_address_name: publicip001
+
+    - name: Delete network interface
+        azure_rm_networkinterface:
+            resource_group: Testing
+            name: nic003
+            state: absent
+'''
+
+RETURN = '''
+changed:
+    description: Whether or not the object was changed.
+    returned: always
+    type: bool
+    sample: True
+check_mode:
+    description: Whether or not the module was executed in check mode.
+    returned: always
+    type: bool
+    sample: True
+Results:
```

Comment:
results is already a reserved key for Ansible when using loops, this might be confusing and/or create conflicts